docs/reference/hooks.md

```diff
@@ -490,52 +492,83 @@ def global_admin_css():
 Add additional JavaScript files or code snippets to the page editor.
 
 ```python
-from django.utils.html import format_html_join
-from django.utils.safestring import mark_safe
+# wagtail_hooks.py
 from django.templatetags.static import static
-
+from django.utils.html import format_html, format_html_join
 from wagtail import hooks
 
-@hooks.register('insert_editor_js')
+@hooks.register("insert_editor_js")
 def editor_js():
     js_files = [
-        'js/fireworks.js', # https://fireworks.js.org
+        'js/fireworks.js', 
+        'js/controllers/fireworks_controller.js',  
     ]
-    js_includes = format_html_join('\n', '<script src="{0}"></script>',
+    return format_html_join(
+        '\n',
+        '<script src="{}"></script>',
         ((static(filename),) for filename in js_files)
     )
-    return js_includes + mark_safe(
-        """
-        <script>
-            window.addEventListener('DOMContentLoaded', (event) => {
-                var container = document.createElement('div');
-                container.style.cssText = 'position: fixed; width: 100%; height: 100%; z-index: 100; top: 0; left: 0; pointer-events: none;';
-                container.id = 'fireworks';
-                document.getElementById('main').prepend(container);
-                var options = { "acceleration": 1.2, "autoresize": true, "mouse": { "click": true, "max": 3 } };
-                var fireworks = new Fireworks(document.getElementById('fireworks'), options);
-                fireworks.start();
-            });
-        </script>
-        """
-    )
```

Comment:
@aayushman-singh I think we probably should keep this as the non-Stimulus example. I know I suggested this but I realise there's a bit more to it compared to what's in this example.

I will just move the existing JS to the external file and push.